src/components/ToggleSwitch.js

```diff
@@ -6,101 +6,69 @@ const Icon = require('./Icon');
-    style: React.PropTypes.object,
-    trueIcon: React.PropTypes.element
+    styles: React.PropTypes.object,
+    trueIcon: React.PropTypes.string
   },
 
   getDefaultProps () {
     return {
       checked: false,
+      falseIcon: 'close-skinny',
       leftLabel: 'Off',
       onToggle () {},
       rightLabel: 'On',
       showLabels: false,
       showIcons: true,
-      style: {}
-    };
-  },
-
-  getInitialState () {
-    return {
-      checked: this.props.checked
+      style: {},
+      trueIcon: 'check-skinny'
     };
   },
 
   _handleToggle (event) {
-    this.setState({
-      checked: !this.state.checked
-    }, () => {
-      this.props.onToggle(this.state.checked, event);
-    });
-  },
-
-  _renderLeftLabel (styles) {
-    if (this.props.showLabels) {
-      return (
-        <span className='left-label' onClick={this._handleToggle} style={[styles.label, !this.state.checked && styles.activeLabel || styles.inactiveLabel]}>{this.props.leftLabel}</span>
-      );
-    } else {
-      return null;
-    }
-  },
-
-  _renderRightLabel (styles) {
-    if (this.props.showLabels) {
-      return (
-        <span className='right-label' onClick={this._handleToggle} style={[styles.label, this.state.checked && styles.activeLabel || styles.inactiveLabel]}>{this.props.rightLabel}</span>
-      );
-    } else {
-      return null;
-    }
-  },
-
-  _renderIcons (styles) {
-    if (!this.props.showIcons) {
-      return null;
-    }
-    const trueIcon = this.props.trueIcon || <Icon className='true-icon' style={Object.assign({}, styles.icon, styles.trueIcon)} type='check-skinny' />;
-    const falseIcon = this.props.falseIcon || <Icon className='false-icon' style={Object.assign({}, styles.icon, styles.falseIcon)} type='close-skinny' />;
-
-    return (
-      <span>
-        {trueIcon} {falseIcon}
-      </span>
-    );
+    this.props.onToggle(event);
   },
 
   render () {
     const styles = this.styles();
 
     return (
       <div className='toggle-switch-component' style={styles.component}>
-        {this._renderLeftLabel(styles)}
+        {this.props.showLabels ? (
+          <div className='left-label' onClick={this._handleToggle} style={Object.assign({}, styles.label, this.props.checked ? styles.inactiveLabel : styles.activeLabel)}>{this.props.leftLabel}</div>
+        ) : null}
         <div
           className='toggle-switch-track'
           onClick={this._handleToggle}
-          style={Object.assign({}, styles.track, styles[this.state.checked + 'Track'])}
+          style={Object.assign({}, styles.track, styles[this.props.checked + 'Track'])}
         >
-          <div className='toggle-switch-toggle' style={Object.assign({}, styles.toggle, styles[this.state.checked + 'Toggle'])}></div>
-          {this._renderIcons(styles)}
+          {this.props.showIcons ? (
+            <Icon className='false-icon' style={Object.assign({}, styles.icon, styles.trueIcon)} type={this.props.trueIcon} />
```

Comment:
The className should be true-icon. I'm also not sure they both need to be wrapped in showIcons conditional as the conditional is the same and there is no element between them.